packages/gatsby-mdx/gatsby-node.js

```diff
@@ -22,27 +32,65 @@ exports.onCreatePage = require("./gatsby/on-create-page");
  */
 exports.onCreateWebpackConfig = require("./gatsby/create-webpack-config");
 
+/**
+ * Create the cache directories
+ */
+
+exports.onPreBootstrap = () => {
+  mkdirp.sync(MDX_WRAPPERS_LOCATION);
+  mkdirp.sync(MDX_SCOPES_LOCATION);
+};
+
+/**
+ * Make graphql function glboal for the static and page loaders
+ */
+exports.createPages = ({ graphql }) => {
+  global.graphql = graphql;
+};
+
 /**
  * Add the MDX extensions as resolvable. This is how the page creator
  * determines which files in the pages/ directory get built as pages.
  */
 exports.resolvableExtensions = (data, pluginOptions) =>
   defaultOptions(pluginOptions).extensions;
 
-/**
- * Convert MDX to JSX so that Gatsby can extract the GraphQL queries.
- */
 exports.preprocessSource = async function preprocessSource(
   { filename, contents },
   pluginOptions
 ) {
   const { extensions, ...options } = defaultOptions(pluginOptions);
   const ext = path.extname(filename);
 
+  /**
+   * Convert MDX to JSX so that Gatsby can extract the GraphQL queries.
+   */
   if (extensions.includes(ext)) {
     const code = await mdx(contents, options);
     return code;
   }
+
+  /**
+   * Intercept MDX wrappers and pass through the original component so the
+   * query can be extracted correctly
+   */
+  if (contents.startsWith(WRAPPER_START)) {
+    const [, originalFile] = contents.replace(WRAPPER_START, "").split("\n// ");
+
+    const code = await fs.readFile(originalFile, "utf8");
+
+    const transpiled = first(
+      compact(
+        await apiRunnerNode(`preprocessSource`, {
```

Comment:
Actually throw the original component through the `preprocessSource` in case it was a compile-to-js file.